src/components/mods/RandomizeWildPokemonMod.ts

```diff
@@ -0,0 +1,44 @@
+import { Component } from "eightbittr/lib/Component";
+import { ICallbackRegister, IMod } from "modattachr/lib/IModAttachr";
+
+import { FullScreenPokemon } from "../../FullScreenPokemon";
+import { IWildPokemonSchema } from "../Maps";
+
+/**
+ * Mod that randomizes Pokemon encounters based on the original wild Pokemon's type(s).
+ */
+export class RandomizeWildPokemonMod<TGameStartr extends FullScreenPokemon> extends Component<TGameStartr> implements IMod {
+    /**
+     * Name of the mod.
+     */
+    public readonly name: string = "Randomize Wild Pokemon";
+
+    /**
+     * Mod events, keyed by name.
+     */
+    public readonly events: ICallbackRegister = {
+        onRandomizePokemon: (chosen: IWildPokemonSchema): IWildPokemonSchema => {
+            const pokemonName: string = chosen.title.join("");
+            const pokemonTypes: string[] = [];
```

Comment:
Do you want me to use this.gameStarter.constants.pokemon.byName.types instead? Also now looking at it I could get away with not doing the for loop if I do:
 const pokemonTypes: string[] = this.gameStarter.constants.pokemon.byName[pokemonName].types;